static/app/components/carousel.tsx

```diff
@@ -0,0 +1,148 @@
+import {useEffect, useRef, useState} from 'react';
+import styled from '@emotion/styled';
+
+import {Button} from 'sentry/components/button';
+import {IconArrow} from 'sentry/icons';
+
+type Props = {
+  children?: React.ReactNode;
+};
+
+function Carousel({children}: Props) {
+  const ref = useRef<HTMLDivElement | null>(null);
+  const [isAtStart, setIsAtStart] = useState<boolean>(true);
+  const [isAtEnd, setIsAtEnd] = useState<boolean>(false);
+  const [showArrows, setShowArrows] = useState<boolean>(true);
+
+  const setStartAndEnd = (scrollLeft: number, totalScrollDist: number) => {
+    // scrollLeft: the amount scrolled horizontally.
+    // start at 0 and end at totalScrollDist
+
+    // totalScrollDist = scrollWidth - clientWidth
+    // scrollWidth: total element width including overflow
+    // clientWidth: width of the element excluding overflow
+    if (scrollLeft <= 0) {
+      setIsAtStart(true);
+      setIsAtEnd(false);
+    } else if (scrollLeft >= totalScrollDist) {
+      setIsAtStart(false);
+      setIsAtEnd(true);
+    } else {
+      setIsAtStart(false);
+      setIsAtEnd(false);
+    }
+  };
+
+  useEffect(() => {
+    if (!ref.current) {
+      return;
+    }
+    const contentWidth = ref.current.scrollWidth - ref.current.clientWidth;
+    if (contentWidth === 0) {
+      setShowArrows(false);
+    }
+
+    const scrollLeft = ref.current.scrollLeft;
+    setStartAndEnd(scrollLeft, contentWidth);
+  }, [ref]);
```

Comment:
Ref objects shouldn't be used as dependencies since they will never change. The point of a ref is that the object is completely stable (it will never change reference), the `.current` property is mutated inside of it.

So no need to have it as a dependency here